home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/general/email/SendCooperationEmailService.java

```diff
@@ -31,13 +38,20 @@ public void executeAllInvitationsByType() {
     @Override
     protected MailDto createMailDto(InvitationDto invitationDto) {
         CooperationInvitationDto invitation = mapper.convert(invitationDto, CooperationInvitationDto.class);
+
+        Cooperation coop = cooperationRepository.findById(invitation.getCooperationId())
```

Comment:
Is it under job processing? If it is I suggest throwing some HomeProcessingException that will be checked one and handle it on caller level moving the invitation in ERRED state.

Reply:
I can`t find any mention of throwing exceptions except Illegal Argument Exception In case of null id param.